### PR TITLE
feat(rest): endpoint to link sourceProject to list of projects.

### LINKED ISSUE #2230: Endpoint to link Project to other projects.
### Description

This new endpoint will link the source project to other projects which will be passed as a list of array in request body.

Endpoint: http://localhost:8080/resource/api/projects/{sourceProjId}/linkProjects
In body pass list of projectIds.

### PR DESCRIPTION
**Pull request by rudra-superrr**
Issue: Closes #2230 

### How To Test?

Run the endpoint: 
http://localhost:8080/resource/api/projects/{id}/linkProjects
Pass an array of project ids in request body.

Possible responses can be:
image

### CODE REVIEW

**Comment by keerthi-bl** [comment]
I am testing this PR.

**Comment by keerthi-bl** [comment]
Test Successfull.

image

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -631,6 +635,99 @@ public ResponseEntity linkReleases(
...
+
+    try {
+
+        for(String projId: projectIdsInRequestBody) {
+            Project proj = projectService.getProjectForUserById(projId, sw360User);
+            Map<String, ProjectProjectRelationship> linkedProject= proj.getLinkedProjects();
+
+            if (proj.getLinkedProjects().keySet().contains(id)) {
+                alreadyLinkedIds.add(projId);
+                responseMap.put("Message regarding already linked project(s)", "Project ids are: " + alreadyLinkedIds);
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:678]: Move 678 - 680 out of for loop
Like this
```
if (!alreadyLInkedIds.isEmty()) {
  responseMap.put("Message regarding already linked project(s)", "Project ids are: " + alreadyLinkedIds);
  status = HttpStatus.CONFLICT;
  projAlreadyLinkedIds = true;
}
```

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 2]
```diff
@@ -631,6 +635,99 @@ public ResponseEntity linkReleases(
...
+                status = HttpStatus.CONFLICT;
+                projAlreadyLinkedIds = true;
+                continue;
+            }
+
+            if (!PermissionUtils.makePermission(proj, sw360User).isActionAllowed(RequestedAction.WRITE)) {
+                idsSentToModerator.add(projId);
+                responseMap.put("Message regarding project(s) sent to Moderator", "Project ids are: " + idsSentToModerator);
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:686]: Move out of for loop

Comment (hoangnt2) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:686]: Maybe this missing
`continue;`

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 3]
```diff
@@ -631,6 +635,99 @@ public ResponseEntity linkReleases(
...
+                status = HttpStatus.ACCEPTED;
+            }
+        }
+
+        if ((linkedProjIds && projAlreadyLinkedIds) || (projIdsWithCyclicPath && projAlreadyLinkedIds) || (linkedProjIds && projIdsWithCyclicPath)
+            || (linkedProjIds && projIdsSentToModerator) || (projAlreadyLinkedIds && projIdsSentToModerator) || (projIdsWithCyclicPath && projIdsSentToModerator)) {
+            status = HttpStatus.CONFLICT;
+        }
+
+    }catch (TException e) {
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:721]:
```suggestion
    } catch (TException e) {
```

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 4]
```diff
@@ -631,6 +635,99 @@ public ResponseEntity linkReleases(
...
+                status = HttpStatus.CREATED;
+            }
+
+            if (updatedstatus == RequestStatus.SENT_TO_MODERATOR) {
+                projIdsSentToModerator = true;
+                status = HttpStatus.ACCEPTED;
+            }
+        }
+
+        if ((linkedProjIds && projAlreadyLinkedIds) || (projIdsWithCyclicPath && projAlreadyLinkedIds) || (linkedProjIds && projIdsWithCyclicPath)
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:716]: If the condition is complex, can you simplify it

**Comment by rudra-superrr** [comment]
Hi @hoangnt2 , thanks for reviewing.
Made the requested changes, please look into the changes.

**Comment by hoangnt2** [comment]
LGTM

**Pull request merged**